common/service-support-cdi/src/main/java/io/helidon/common/servicesupport/cdi/AnnotationLookupResult.java

```diff
@@ -0,0 +1,130 @@
+        // First check annotation on element
+        A annotation = element.getAnnotation(annotClass);
+        if (annotation != null) {
+            return new AnnotationLookupResult<>(AnnotationSiteType.METHOD, annotation);
+        }
+        // Finally check annotations on class
+        annotation = element.getDeclaringClass()
+                .getAnnotation(annotClass);
+        if (annotation == null) {
+            annotation = clazz.getAnnotation(annotClass);
+        }
+        return annotation == null ? null : new AnnotationLookupResult<>(AnnotationSiteType.CLASS, annotation);
+    }
+
+    /**
+     * Creates a collection of lookup results given an annotation type and an {@code Annotated} site.
+     *
+     * @param annotated  the annotated site
+     * @param annotClass the annotation type class
+     * @param <A>        the annotation type
+     * @return lookup result
+     */
+    public static <A extends Annotation> Collection<AnnotationLookupResult<A>> lookupAnnotations(Annotated annotated,
+            Class<A> annotClass) {
+        // We have to filter by annotation class ourselves, because annotatedMethod.getAnnotations(Class) delegates
+        // to the Java method. That would bypass any annotations that had been added dynamically to the configurator.
+        return annotated.getAnnotations()
+                .stream()
+                .filter(annotClass::isInstance)
+                .map(annotation -> new AnnotationLookupResult<>(AnnotationSiteType.matchingType(annotated),
+                        annotClass.cast(annotation)))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Creates a collection of lookup results for an annotation on a given method or, failing that, at the corresponding class
+     * level.
+     *
+     * @param annotatedType   the type on which the annotation might appear
+     * @param annotatedMethod the method to check for the annotation
+     * @param annotClass      the annotation to check for
+     * @param <A>             the type of the annotation
+     * @return {@code List} of {@code AnnotationLookupResult}
+     */
+    public static <A extends Annotation> Collection<AnnotationLookupResult<A>> lookupAnnotations(
+            AnnotatedType<?> annotatedType,
+            AnnotatedMethod<?> annotatedMethod,
+            Class<A> annotClass) {
+        Collection<AnnotationLookupResult<A>> result = lookupAnnotations(annotatedMethod, annotClass);
+        if (result.isEmpty()) {
+            result = lookupAnnotations(annotatedType, annotClass);
+        }
+        return result;
+    }
+
+    private AnnotationLookupResult(AnnotationSiteType type, A annotation) {
+        this.type = type;
+        this.annotation = annotation;
+    }
+
+    /**
+     * Returns the site type where the annotation is located.
+     *
+     * @return the matching type
+     */
+    public AnnotationSiteType getType() {
```

Comment:
Do not use `get`, also maybe `siteType()` may be a better name?

Reply:
Fixed